src/components/feedbackForm/FeedbackForm.jsx

```diff
@@ -0,0 +1,163 @@
+import classes from './feedbackForm.scss';
+import { useEffect, useState } from 'react';
+import { useHistory } from 'react-router-dom';
+import DataService from '../../dataService/DataService';
+
+const EMAIL_REGEXP = /^([A-Za-z0-9_\-\.])+\@([A-Za-z0-9_\-\.])+\.([A-Za-z]{2,4})$/;
+
+const VALIDATORS = {
+  isEmpty: value => !!(value.length === 0),
+  minLength: (value, param) => !!(value.length < param),
+  maxLength: (value, param) => !!(value.length > param),
+  email: value => !EMAIL_REGEXP.test(value)
+};
+
+const ERROR_MESSAGES = {
+  isEmpty: 'This field is required',
+  minLength: 'Incorrect length',
+  maxLength: 'Incorrect length',
+  email: 'Incorrect email'
+};
+
+function useValidation(value, validators) {
```

Comment:
just an advice - custom hooks can be moved in separate files to make your component file cleaner